hugo/content/docs/recipes/validation/dependency-loops.md

```diff
@@ -0,0 +1,258 @@
+---
+title: "Dependency loops"
+weight: 100
+---
+
+## What is the problem?
```

Comment:
My personal impression is, that the motivation is not clear enough. I write some ideas in order to clarify this impression:

What about supporting some examples with more details, e.g. the Langium grammar for dependencies

```langium
Component: name=ID ('depends on' dependencies+=[Component]*)?;
```

or a class diagram for the resulting TS structure

```mermaid
classDiagram
class Component {
dependencies: Component[*]
}
```

?

The purpose, i.e. to create errors for the users, is missing.

List some more examples and name this examples more concrete:
- no cyclic data flow in flow charts
- no cyclic imports of dependencies
- no cycles in inheritance hierarchies of data types/types/classes
- no cycles in function calls (might not be supported by some architectures)

Reply:
I added some concrete examples